### PR TITLE
#81: Fix merge-once Project approval and dirty PR routing

### LINKED ISSUE #81: Fix merge-once Project approval and dirty PR routing
## Issue Setup

- UAT Required: No
- Related Parent Issue or Context: Live dogfood merge pass for #63, #64, #65, and #74

## Issue Goal

Fix `merge-once` so Project `Merging` is treated as the workflow approval signal, and dirty PRs are routed to Rework before missing GitHub review metadata can mask mergeability blockers.

## Why Now

Live dogfood merging routed #63, #64, #65, and #74 away from the merge lane because GitHub `reviewDecision` was empty, even though Project status was already `Merging`. For dirty PRs (#63/#64), this also hid the actionable `DIRTY` state and sent them to `Need Human Input` instead of `Rework`.

## Issue Context

Jade Symphony uses GitHub Project v2 project #9 as the workflow state machine. Bootstrap workflow semantics define `Merging` as the human-approved land state. This repo may not have GitHub PR review objects even when Project status and workpad evidence show that a PR is approved for landing.

### Decisions

- Project `Merging` is the approval source for the merge lane.
- Empty GitHub `reviewDecision` is not by itself a blocker when the issue is already in Project `Merging`.

### Assumptions

- Explicit negative GitHub review signals, failing checks, and dirty mergeability remain blockers.

## Non-Negotiable Guardrails

- Do not implement a continuous merge loop.
- Do not change Agent Review or Human Review role boundaries.
- Do not merge unrelated open PRs in this issue branch.
- Do not edit files under `docs/bootstrap/references/openai-symphony`.

### In Scope

- Update `src/merge_lane.rs` decision ordering so failing/pending checks and mergeability states are not hidden by missing GitHub review metadata.
- Treat an issue already in configured `Merging` status as sufficient approval when GitHub `reviewDecision` is empty.
- Continue blocking explicit negative PR review signals such as changes requested.
- Ensure dirty/behind PRs route to `Rework`.
- Ensure clean Project-Merging PRs with empty GitHub `reviewDecision` can be treated as ready to merge.
- Update README or dogfood readiness docs where they currently imply GitHub review approval is always required.

### Out of Scope

- Continuous merge polling.
- Review Agent behavior changes.
- Human Review state transitions.

### Target Repository / Package

- `Alive24/jade-symphony`

### Relevant Knowledge Sources

- `docs/bootstrap/JADE_WORKFLOW.md`
- `docs/bootstrap/JADE_SYMPHONY_SPEC.md`
- `README.md`
- `docs/dogfood-readiness.md`

### Relevant Code Paths

- `src/merge_lane.rs`

## Current State

`merge-once` checks GitHub `reviewDecision` before mergeability status. When `reviewDecision` is empty, it routes the issue to `Need Human Input` even if the PR is dirty and should be `Rework`, or clean and already approved by Project `Merging`.

## Deliverable Shape

Code and tests that make the merge-lane decision model match Project-state approval semantics, plus a small documentation correction.

## Risks or Constraints

- The change must not allow PRs with explicit negative review signals, failing checks, or dirty merge states to merge.
- The change must preserve the main implementation agent / review agent / human review role boundaries.

## Expected Outcome

`merge-once` can land clean Project-Merging PRs without requiring GitHub review objects, and routes dirty Project-Merging PRs to Rework even when GitHub review metadata is empty.

### Completion Criteria

- Unit tests cover dirty PRs with empty review metadata routing to Rework.
- Unit tests cover clean Project-Merging PRs with empty review metadata being merge-ready.
- Existing approved-review behavior remains supported.
- README or dogfood readiness docs describe Project `Merging` as the approval source for `merge-once`.

### Functional Verification

- `cargo test`
- `cargo fmt --check`
- `cargo clippy --all-targets --all-features -- -D warnings`

### UAT

- Not required.

### Context Verification

- Confirm `docs/bootstrap/JADE_WORKFLOW.md` and `docs/bootstrap/JADE_SYMPHONY_SPEC.md` still describe `Merging` as the land/approval state.

### PR DESCRIPTION
**Pull request by Alive24**
## Summary

- Treat Project `Merging` as the merge-lane approval source when GitHub `reviewDecision` is empty.
- Reorder merge-lane blockers so failing checks and dirty/behind merge states are evaluated before review metadata.
- Keep explicit negative review decisions blocked, routing `CHANGES_REQUESTED` to Rework.
- Update docs to describe Project `Merging` approval semantics.

## Verification

- `cargo test`
- `cargo fmt --check`
- `cargo clippy --all-targets --all-features -- -D warnings`

Closes #81

### CODE REVIEW

**Review by Alive24 — COMMENTED** [review]
## 📋 Review Summary
Skipped due to error: Command '['gemini', '-p', '\nYou are a Code Review Architect. Follow these guidelines strictly:\n---\nname: code-review-commons\ndescription: Common guidelines, persona and critical constraints for performing high-quality code reviews. Use this skill when performing a /code-review or /pr-code-review command.\nuser-invokable: false\n---\n\n# Code Review Commons\n\n## PERSONA\n\nYou are a very experienced **Principal Software Engineer** and a meticulous **Code Review Architect**. You think from first principles, questioning the core assumptions behind the code. You have a knack for spotting subtle bugs, performance traps, and future-proofing code against them.\n\n## OBJECTIVE\n\nYour task is to deeply understand the **intent and context** of the provided code changes (diff content) and then perform a **thorough, actionable, and objective** review.\nYour primary goal is to **identify potential bugs, security vulnerabilities, performance bottlenecks, and clarity issues**.\nProvide **insightful feedback** and **concrete, ready-to-use code suggestions** to maintain high code quality and best practices. Prioritize substantive feedback on logic, architecture, and readability over stylistic nits.\n\n## Instructions\n\n1. **Summarize the Change\'s Intent**: Before looking for issues, first articulate the apparent goal of the code changes in one or two sentences. Use this understanding to frame your review.\n2. **Establish context** by reading relevant files. Prioritize:\n    a. All files present in the diff.\n    b. Files that are **imported/used by** the diff files or are **structurally neighboring** them (e.g., related configuration or test files).\n3. **Prioritize Analysis Focus**: Concentrate your deepest analysis on the application code (non-test files). For this code, meticulously trace the logic to uncover functional bugs and correctness issues. Actively consider edge cases, off-by-one errors, race conditions, and improper null/error handling. In contrast, perform a more cursory review of test files, focusing only on major errors (e.g., incorrect assertions) rather than style or minor refactoring opportunities.\n4. **Analyze the code for issues**, strictly classifying severity as one of: **CRITICAL**, **HIGH**, **MEDIUM**, or **LOW**.\n\n## Critical Constraints\n\n**STRICTLY follow these rules for review comments:**\n\n* **Location:** You **MUST** only provide comments on lines that represent actual changes in the diff. This means your comments must refer **only to lines beginning with `+` or `-`**. **DO NOT** comment on context lines (lines starting with a space).\n* **Relevance:** You **MUST** only add a review comment if there is a demonstrable **BUG**, **ISSUE**, or a significant **OPPORTUNITY FOR IMPROVEMENT** in the code changes.\n* **Tone/Content:** **DO NOT** add comments that:\n    * Tell the user to "check," "confirm," "verify," or "ensure" something.\n    * Explain what the code change does or validate its purpose.\n    * Explain the code to the author (they are assumed to know their own code).\n    * Comment on missing trailing newlines or other purely stylistic issues that do not affect code execution or readability in a meaningful way.\n* **Substance First:** **ALWAYS** prioritize your analysis on the **correctness** of the logic, the **efficiency** of the implementation, and the **long-term maintainability** of the code.\n* **Technical Detail:**\n    * Pay **meticulous attention to line numbers and indentation** in code suggestions; they **must** be correct and match the surrounding code.\n    * **NEVER** comment on license headers, copyright headers, or anything related to future dates/versions (e.g., "this date is in the future").\n* **Formatting/Structure:**\n    * Keep the **change summary** concise (aim for a single sentence).\n    * Keep **comment bodies concise** and focused on a single issue.\n    * If a similar issue exists in **multiple locations**, state it once and indicate the other locations instead of repeating the full comment.\n    * **AVOID** mentioning your instructions, settings, or criteria in the final output.\n\n**Severity Guidelines (for consistent classification):**\n\n* **Functional correctness bugs that lead to behavior contrary to the change\'s intent should generally be classified as HIGH or CRITICAL.**\n* **CRITICAL:** Security vulnerabilities, system-breaking bugs, complete logic failure.\n* **HIGH:** Performance bottlenecks (e.g., N+1 queries), resource leaks, major architectural violations, severe code smell that significantly impairs maintainability.\n* **MEDIUM:** Typographical errors in code (not comments), missing input validation, complex logic that could be simplified, non-compliant style guide issues (e.g., wrong naming convention).\n* **LOW:** Refactoring hardcoded values to constants, minor log message enhancements, comments on docstring/Javadoc expansion, typos in documentation (.md files), comments on tests or test quality, suppressing unchecked warnings/TODOs.\n\n\nReview this PR.\nTitle: #81: Fix merge-once Project approval and dirty PR routing\nBody: ## Summary\n\n- Treat Project `Merging` as the merge-lane approval source when GitHub `reviewDecision` is empty.\n- Reorder merge-lane blockers so failing checks and dirty/behind merge states are evaluated before review metadata.\n- Keep explicit negative review decisions blocked, routing `CHANGES_REQUESTED` to Rework.\n- Update docs to describe Project `Merging` approval semantics.\n\n## Verification\n\n- `cargo test`\n- `cargo fmt --check`\n- `cargo clippy --all-targets --all-features -- -D warnings`\n\nCloses #81\n\n\nDiff:\n```diff\ndiff --git a/README.md b/README.md\nindex cc0ad6a..415d315 100644\n--- a/README.md\n+++ b/README.md\n@@ -50,8 +50,9 @@ worker supervision are still future work.\n   Human Review remains valid.\n - `merge-once` can consume issues already in `Merging`, resolve a single linked\n   PR, run guarded GitHub PR preflight checks, write durable workpad evidence,\n-  merge clean approved PRs with explicit `--write`, and route blockers to\n-  `Rework` or `Need Human Input` without ever setting `Human Review`.\n+  treat Project `Merging` as the approval signal for clean PRs, merge with\n+  explicit `--write`, and route blockers to `Rework` or `Need Human Input`\n+  without ever setting `Human Review`.\n - structured Rework diagnostics can render compact, durable issue workpad\n   evidence for confirmed review findings, merge conflicts, dirty PRs,\n   validation failures, and runtime failures before a transition to `Rework`.\ndiff --git a/docs/dogfood-readiness.md b/docs/dogfood-readiness.md\nindex 305d110..b1ad204 100644\n--- a/docs/dogfood-readiness.md\n+++ b/docs/dogfood-readiness.md\n@@ -23,7 +23,7 @@ yet.\n | Execution profiles | First-slice profile discovery exists. Workflow config can point to a cockpit-tools Codex `codex_instances.json` file, and Jade treats each instance `name` as a profile/worker identity while ignoring account binding fields. If the cockpit-tools file is missing, explicit `profiles.entries` are used. This is not a full account manager. |\n | Agent backends | Dry-run backend plus conservative Codex and Claude Code subprocess backends exist. Prepared runs include selected profile/instance metadata and profile environment context. Full Codex app-server and Claude Code protocol parity are not implemented yet. |\n | Agent Review | Finding classes, fake reviewer lifecycle, Gemini CLI subprocess backend, role-bound transition decisions, evidence-first Rework diagnostics for confirmed findings, review-freshness evidence for Merging conflict repair, bounded `review-loop` selection/reconciliation, and workpad/status evidence helpers exist. Persistent background review worker supervision is not implemented yet. |\n-| Merging | `merge-once` can inspect issues already in `Merging`, require one linked PR, check live GitHub PR state/review/check/mergeability data where available, merge clean approved PRs only with explicit `--write`, and route blockers to `Rework` or `Need Human Input` with workpad evidence. Continuous merge polling and issue closing beyond Project `Done` are not implemented yet. |\n+| Merging | `merge-once` can inspect issues already in `Merging`, require one linked PR, treat Project `Merging` as the approval signal, check live GitHub PR state/review/check/mergeability data where available, merge clean PRs only with explicit `--write`, and route blockers to `Rework` or `Need Human Input` with workpad evidence. Continuous merge polling and issue closing beyond Project `Done` are not implemented yet. |\n | Project doctor | Read-only `doctor` / `audit-project` reports workflow invariant violations from normalized tracker issues, including missing PR handoff evidence, missing review pass evidence, dirty Merging PRs, missing runtime ownership hints, and queued issues with PRs. Repair mode is not implemented yet. |\n | Observability | Operator-readable terminal snapshots report polling, running, retrying, skipped issues, gate details, token counters, event-log path, and integration gaps. JSONL event-log primitives exist and `run-once` writes dry-run events with actor and profile metadata. Runtime state files are written during write-mode `run-loop` issue execution, including actor role/label, git author, and optional profile/instance identity when configured; resume, retry, usage-limit pause, and stall supervision events are also recorded. No web/API surface yet. |\n | Usage-limit pause/resume | Conservative usage-limit/rate-limit/resource-exhausted classification exists for subprocess agent events and review job output. `run-loop` records usage-limit pauses in workpad/runtime retry state and does not advance to `Agent Review`; review workpads surface usage-limit failures without moving to `Human Review`. Vendor-specific quota management is not implemented. |\ndiff --git a/src/merge_lane.rs b/src/merge_lane.rs\nindex 9f976f5..6d9beb7 100644\n--- a/src/merge_lane.rs\n+++ b/src/merge_lane.rs\n@@ -166,22 +166,6 @@ pub fn merge_lane_decision(\n         };\n     }\n \n-    if !matches!(\n-        status.review_decision.as_deref(),\n-        Some("APPROVED") | Some("approved")\n-    ) {\n-        return MergeLaneDecision {\n-            kind: MergeLaneDecisionKind::ReviewNotApproved,\n-            issue_ref: issue.identifier.clone(),\n-            pr_url: Some(status.url.clone()),\n-            target_state: Some("need_human_input"),\n-            reason: format!(\n-                "pull request review decision is `{}`",\n-                status.review_decision.as_deref().unwrap_or("missing")\n-            ),\n-        };\n-    }\n-\n     if let Some(failing) = first_failing_check(&status.checks) {\n         return MergeLaneDecision {\n             kind: MergeLaneDecisionKind::ChecksFailing,\n@@ -203,38 +187,74 @@ pub fn merge_lane_decision(\n     }\n \n     match status.merge_state_status.as_deref() {\n-        Some("CLEAN") | Some("HAS_HOOKS") => MergeLaneDecision {\n-            kind: MergeLaneDecisionKind::ReadyToMerge,\n-            issue_ref: issue.identifier.clone(),\n-            pr_url: Some(status.url.clone()),\n-            target_state: Some("done"),\n-            reason: "pull request passed merge preflight".into(),\n-        },\n-        Some("DIRTY") | Some("BEHIND") => MergeLaneDecision {\n-            kind: MergeLaneDecisionKind::MergeDirty,\n-            issue_ref: issue.identifier.clone(),\n-            pr_url: Some(status.url.clone()),\n-            target_state: Some("rework"),\n-            reason: format!(\n-                "pull request merge state is `{}`",\n-                status.merge_state_status.as_deref().unwrap_or_default()\n-            ),\n-        },\n-        Some(other) => MergeLaneDecision {\n-            kind: MergeLaneDecisionKind::MergeabilityUnknown,\n-            issue_ref: issue.identifier.clone(),\n-            pr_url: Some(status.url.clone()),\n-            target_state: Some("need_human_input"),\n-            reason: format!("pull request merge state is `{other}`"),\n-        },\n-        None => MergeLaneDecision {\n-            kind: MergeLaneDecisionKind::MergeabilityUnknown,\n+        Some("DIRTY") | Some("BEHIND") => {\n+            return MergeLaneDecision {\n+                kind: MergeLaneDecisionKind::MergeDirty,\n+                issue_ref: issue.identifier.clone(),\n+                pr_url: Some(status.url.clone()),\n+                target_state: Some("rework"),\n+                reason: format!(\n+                    "pull request merge state is `{}`",\n+                    status.merge_state_status.as_deref().unwrap_or_default()\n+                ),\n+            };\n+        }\n+        Some("CLEAN") | Some("HAS_HOOKS") => {}\n+        Some(other) => {\n+            return MergeLaneDecision {\n+                kind: MergeLaneDecisionKind::MergeabilityUnknown,\n+                issue_ref: issue.identifier.clone(),\n+                pr_url: Some(status.url.clone()),\n+                target_state: Some("need_human_input"),\n+                reason: format!("pull request merge state is `{other}`"),\n+            };\n+        }\n+        None => {\n+            return MergeLaneDecision {\n+                kind: MergeLaneDecisionKind::MergeabilityUnknown,\n+                issue_ref: issue.identifier.clone(),\n+                pr_url: Some(status.url.clone()),\n+                target_state: Some("need_human_input"),\n+                reason: "pull request merge state is missing".into(),\n+            };\n+        }\n+    }\n+\n+    if let Some((target_state, reason)) = blocking_review_decision(&status.review_decision) {\n+        return MergeLaneDecision {\n+            kind: MergeLaneDecisionKind::ReviewNotApproved,\n             issue_ref: issue.identifier.clone(),\n             pr_url: Some(status.url.clone()),\n-            target_state: Some("need_human_input"),\n-            reason: "pull request merge state is missing".into(),\n-        },\n+            target_state: Some(target_state),\n+            reason,\n+        };\n     }\n+\n+    MergeLaneDecision {\n+        kind: MergeLaneDecisionKind::ReadyToMerge,\n+        issue_ref: issue.identifier.clone(),\n+        pr_url: Some(status.url.clone()),\n+        target_state: Some("done"),\n+        reason: "pull request passed merge preflight with Project Merging approval".into(),\n+    }\n+}\n+\n+fn blocking_review_decision(review_decision: &Option<String>) -> Option<(&\'static str, String)> {\n+    let decision = review_decision.as_deref().unwrap_or_default().trim();\n+    if decision.is_empty() || decision.eq_ignore_ascii_case("APPROVED") {\n+        return None;\n+    }\n+\n+    let target_state = if decision.eq_ignore_ascii_case("CHANGES_REQUESTED") {\n+        "rework"\n+    } else {\n+        "need_human_input"\n+    };\n+\n+    Some((\n+        target_state,\n+        format!("pull request review decision is `{decision}`"),\n+    ))\n }\n \n pub fn pull_request_status_from_linked(\n@@ -508,6 +528,23 @@ mod tests {\n         assert_eq!(decision.target_state, Some("done"));\n     }\n \n+    #[test]\n+    fn clean_project_merging_pr_without_github_review_decision_is_ready_to_merge() {\n+        let issue = issue("Merging", vec![pr()]);\n+        let mut status = clean_status();\n+        status.review_decision = Some(String::new());\n+        let decision = merge_lane_decision(\n+            &issue,\n+            "Merging",\n+            "main",\n+            &issue.linked_pull_requests,\n+            Some(&status),\n+        );\n+\n+        assert_eq!(decision.kind, MergeLaneDecisionKind::ReadyToMerge);\n+        assert_eq!(decision.target_state, Some("done"));\n+    }\n+\n     #[test]\n     fn dirty_pr_routes_to_rework() {\n         let issue = issue("Merging", vec![pr()]);\n@@ -525,6 +562,41 @@ mod tests {\n         assert_eq!(decision.target_state, Some("rework"));\n     }\n \n+    #[test]\n+    fn dirty_pr_without_github_review_decision_routes_to_rework() {\n+        let issue = issue("Merging", vec![pr()]);\n+        let mut status = clean_status();\n+        status.merge_state_status = Some("DIRTY".into());\n+        status.review_decision = Some(String::new());\n+        let decision = merge_lane_decision(\n+            &issue,\n+            "Merging",\n+            "main",\n+            &issue.linked_pull_requests,\n+            Some(&status),\n+        );\n+\n+        assert_eq!(decision.kind, MergeLaneDecisionKind::MergeDirty);\n+        assert_eq!(decision.target_state, Some("rework"));\n+    }\n+\n+    #[test]\n+    fn changes_requested_review_routes_to_rework() {\n+        let issue = issue("Merging", vec![pr()]);\n+        let mut status = clean_status();\n+        status.review_decision = Some("CHANGES_REQUESTED".into());\n+        let decision = merge_lane_decision(\n+            &issue,\n+            "Merging",\n+            "main",\n+            &issue.linked_pull_requests,\n+            Some(&status),\n+        );\n+\n+        assert_eq!(decision.kind, MergeLaneDecisionKind::ReviewNotApproved);\n+        assert_eq!(decision.target_state, Some("rework"));\n+    }\n+\n     #[test]\n     fn missing_pr_routes_to_human_input() {\n         let issue = issue("Merging", Vec::new());\n\n```\n\nWrite a review markdown file exactly in this format:\n## 📋 Review Summary\n[2-3 sentences summarizing the PR]\n## 🔍 General Feedback\n- [Feedback point 1]\n- [Feedback point 2]\n<COMMENT>\n[SEVERITY] [COMMENT_TEXT]\n</COMMENT>\n\n(Only add <COMMENT> if there\'s inline feedback). Do not include any other text before or after.\n']' timed out after 60 seconds
## 🔍 General Feedback
- None

**Review by Alive24 — COMMENTED** [review]
## 📋 Review Summary

This PR resolves an important issue in the merge-lane by correctly handling Project approval status and appropriately routing dirty PRs.

## 🔍 General Feedback

- **Safer Handoffs**: Correctly routing dirty PRs (like those with conflicts or failing checks) prevents the merge lane from stalling or blindly attempting invalid merges.
- **Accurate Preflights**: Ensuring the Project-level approval status is properly incorporated into the merge decision solidifies the authority boundary before a merge is executed.
